src/apps/bridge/bridgePowerController.cpp

```diff
@@ -197,13 +197,15 @@ void BridgePowerController::_update(void) {
             time_to_sleep_ms = MAX(subsampleTimeRemainingS * 1000, MIN_TASK_SLEEP_MS);
             break;
           } else {
-            uint32_t nextSubsampleEpochS = _subsampleIntervalStartS + _subsampleIntervalS;
+            const uint32_t nextSubsampleEpochS = _subsampleIntervalStartS + _subsampleIntervalS;
             _subsampleIntervalStartS = nextSubsampleEpochS;
-            stateLogPrintTarget("Subsampling Off", nextSubsampleEpochS);
-            time_to_sleep_ms =
-                (currentCycleS < nextSubsampleEpochS)
-                    ? MAX((nextSubsampleEpochS - currentCycleS) * 1000, MIN_TASK_SLEEP_MS)
-                    : MIN_TASK_SLEEP_MS;
+            const uint32_t secondsUntilNextSubsample =
+                nextSubsampleEpochS > currentCycleS ? nextSubsampleEpochS - currentCycleS : 0;
+            // Check whether this is the last subsample within a sample duration
+            // If so, only sleep until sampling off, to align the next sample
+            const uint32_t timeToSleepS = MIN(sampleTimeRemainingS, secondsUntilNextSubsample);
```

Comment:
The crux of the fix

Reply:
@towynlin one thing I don't understand - why did we only see the glitch every 10 hours or so?

Reply:
Agreed that it's odd. The only explanation I can think of is that most of the time the power controller task must wake up slightly early with respect to the RTC goal. In that case the extra 20-minute gap bug doesn't manifest. If there were no clock drift and sleep durations were perfectly precise then there would be a 20-minute gap alternating with a 20-minute sampling period.